Makefile

```
@@ -414,31 +425,44 @@ androidupdate: all
 	cp $(ANDROID_LAUNCHER_DIR)/bin/NativeActivity.apk \
 		koreader-android-$(ANDROID_ARCH)$(KODEDUG_SUFFIX)-$(VERSION).apk
 
-debianupdate: all
+LINUX_PACKAGE:=koreader-linux-$(LINUX_ARCH_NAME)$(KODEDUG_SUFFIX)-$(VERSION).tar.xz
+linuxupdate: all
 	mkdir -pv \
-		$(INSTALL_DIR)/debian/usr/bin \
-		$(INSTALL_DIR)/debian/usr/lib \
-		$(INSTALL_DIR)/debian/usr/share/pixmaps \
-		$(INSTALL_DIR)/debian/usr/share/applications \
-		$(INSTALL_DIR)/debian/usr/share/doc/koreader \
-		$(INSTALL_DIR)/debian/usr/share/man/man1
-
-	cp -pv resources/koreader.png $(INSTALL_DIR)/debian/usr/share/pixmaps
-	cp -pv $(DEBIAN_DIR)/koreader.desktop $(INSTALL_DIR)/debian/usr/share/applications
-	cp -pv $(DEBIAN_DIR)/copyright COPYING $(INSTALL_DIR)/debian/usr/share/doc/koreader
-	cp -pv $(DEBIAN_DIR)/koreader.sh $(INSTALL_DIR)/debian/usr/bin/koreader
-	cp -Lr $(INSTALL_DIR)/koreader $(INSTALL_DIR)/debian/usr/lib
-
-	gzip -cn9 $(DEBIAN_DIR)/changelog > $(INSTALL_DIR)/debian/usr/share/doc/koreader/changelog.Debian.gz
-	gzip -cn9 $(DEBIAN_DIR)/koreader.1 > $(INSTALL_DIR)/debian/usr/share/man/man1/koreader.1.gz
+		$(INSTALL_DIR)/linux/usr/bin \
+		$(INSTALL_DIR)/linux/usr/lib \
+		$(INSTALL_DIR)/linux/usr/share/pixmaps \
+		$(INSTALL_DIR)/linux/usr/share/applications \
+		$(INSTALL_DIR)/linux/usr/share/doc/koreader \
+		$(INSTALL_DIR)/linux/usr/share/man/man1
+
+	cp -pv resources/koreader.png $(INSTALL_DIR)/linux/usr/share/pixmaps
+	cp -pv $(LINUX_DIR)/koreader.desktop $(INSTALL_DIR)/linux/usr/share/applications
+	cp -pv $(LINUX_DIR)/copyright COPYING $(INSTALL_DIR)/linux/usr/share/doc/koreader
+	cp -pv $(LINUX_DIR)/koreader.sh $(INSTALL_DIR)/linux/usr/bin/koreader
+	cp -Lr $(INSTALL_DIR)/koreader $(INSTALL_DIR)/linux/usr/lib
+
+	gzip -cn9 $(LINUX_DIR)/koreader.1 > $(INSTALL_DIR)/linux/usr/share/man/man1/koreader.1.gz
 
 	chmod 644 \
-		$(INSTALL_DIR)/debian/usr/share/doc/koreader/changelog.Debian.gz \
-		$(INSTALL_DIR)/debian/usr/share/doc/koreader/copyright \
-		$(INSTALL_DIR)/debian/usr/share/man/man1/koreader.1.gz
+		$(INSTALL_DIR)/linux/usr/share/doc/koreader/copyright \
+		$(INSTALL_DIR)/linux/usr/share/man/man1/koreader.1.gz
 
 	rm -rf \
-		$(INSTALL_DIR)/debian/usr/lib/koreader/{ota,cache,clipboard,screenshots,spec,tools,resources/fonts,resources/icons/src}
+		$(INSTALL_DIR)/linux/usr/lib/koreader/{ota,cache,clipboard,screenshots,spec,tools,l10n/templates,resources/fonts,resources/icons/src}
+
+	# remove leftovers
+	find $(INSTALL_DIR)/linux -type f \( -name ".git" -o -name ".gitignore" -o -name "discovery2spore" -o -name "wadl2spore" -o -name "*.txt" -o -name "LICENSE*" -o -name "NOTICE" -o -name "README.md" \) -print0 | xargs -0 rm -rf
+	find $(INSTALL_DIR)/linux -type d \( -name "test" -o -name ".github" \) -print0 | xargs -0 rm -rf
+
+	# fix permissions
+	find $(INSTALL_DIR)/linux -type d -print0 | xargs -0 chmod 755
+	find $(INSTALL_DIR)/linux -executable -type f -print0 | xargs -0 chmod 755
```

`find … -executable` will already include directories:
>  -executable
>      Matches files which are executable and directories which are searchable (in a file name resolution sense) by the current user.

But I would just use `chmod -R u=rwX,og=r …`.

---

Could you make a nice suggestion with the recursive chmod instead of find?

---

I figure that's simply supposed to read like this? But let's await confirmation.
```suggestion
	chmod -R u=rwX,og=r $(INSTALL_DIR)/linux
```